src/components/TweetsList.jsx

```diff
@@ -1,79 +1,76 @@
-import * as React from "react"
+import { useState, useEffect } from "react"
 import { Tweet } from "react-static-tweets"
 import { HiChevronDoubleDown } from "react-icons/hi"
-import { fetchTweets } from "~/lib/api"
+import { useTweets } from "~/lib/api"
+import { v4 as uuidv4 } from "uuid"
+import Skeleton from "react-loading-skeleton"
 
-/**
- * @type {React.NamedExoticComponent}
- */
-const TweetsList = React.memo(
-  ({ city: location, resource, data: initialData }) => {
-    const [data, setData] = React.useState(initialData)
+const TweetsList = ({ city: location, resource }) => {
+  const [offset, setOffset] = useState(0)
+  const [limit, setLimit] = useState(20)
+  const { data, error } = useTweets({ location, resource, limit, offset })
 
-    const showMore = async () => {
-      const newTweets = await fetchTweets({
-        location,
-        resource,
-        limit: 20,
-        offset: Math.floor(data.length),
-      })
-      setData((data) => data.concat(newTweets))
-    }
+  if (error) return <div>failed to load</div>
+  if (!data) return <Skeleton count={40} />
 
-    if (!(location && resource)) {
-      // Return Please add location & resource
-      return (
-        <div className="py-4 text-xl font-bold">
-          Please select city and resource
-        </div>
-      )
-    } else if (data.length > 0) {
-      // Tweets
-      return (
-        <>
-          {data.map(({ id: tweetId, url: tweetUrl, status: voteCount }) => {
-            return (
-              <div
-                key={tweetId}
-                className="w-full flex flex-col items-center justify-center space-y-4 my-2 px-2"
-              >
-                <Tweet id={tweetId} />
-              </div>
-            )
-          })}
-          {data.length % 20 == 0 && (
-            <button
-              onClick={showMore}
-              className="bg-indigo-200 text-indigo-700 flex items-center justify-center px-4 py-2 rounded-md gap-2 shadow-md focus:outline-none focus:ring-2 focus:ring-indigo-500"
-              disabled={data.length % 20 !== 0}
+  const showMore = () => {
+    setLimit((limit) => limit + 20)
+    setOffset((offset) => offset + 20)
+  }
+
+  if (!(data && location && resource)) {
+    // Return Please add location & resource
+    return (
+      <div className="py-4 text-xl font-bold">
+        Please select city and resource
+      </div>
+    )
+  } else if (data.length > 0) {
+    // Tweets
+    return (
+      <>
+        {data.map(({ id: tweetId, url: tweetUrl, status: voteCount }) => {
+          return (
+            <div
+              key={uuidv4()}
```

Comment:
tweetId is always unique, why use uuid?

Reply:
That's actually what I was wondering, using regular tweet id cause react to throw no-unique key error in console.

look this is what react throws

![image](https://user-images.githubusercontent.com/55892116/116995030-2ace6100-acf7-11eb-98a7-730d69f16561.png)

So I thought of  just switching to uuid for now since it was already added as dependency in package list.

Reply:
@SwarnimWalavalkar Can we get deduplicating setup on the backend? The api shouldn't be returning duplicate tweets.

Reply:
Here's the current deduping logic, it's using mongoose upsert based on the tweet text and the phone numbers found in the tweet...

https://github.com/covidarmy/backend/blob/main/fetchTweets.js#L181

Reply:
yeah the backend seems to be returning response with duplicate keys

![image](https://user-images.githubusercontent.com/55892116/116999240-f78ed080-acfc-11eb-8332-4f62219b9f57.png)

Reply:
@arn4v we should just go with uuid until this is fixed. react only needs the keys to be unique and not changing, uuid fulfil this parameter more than enough.

Reply:
Or we can just use _id in the frontend rather than id (but yeah there should be restrictions on id field if we have one)?